content_sync/pipelines/definitions/concourse/common/steps.py

```diff
@@ -0,0 +1,248 @@
+import json
+from urllib.parse import urljoin
+
+from django.conf import settings
+from ol_concourse.lib.models.pipeline import (
+    Command,
+    DoStep,
+    GetStep,
+    Identifier,
+    PutStep,
+    Step,
+    StepModifierMixin,
+    TaskConfig,
+    TaskStep,
+    TryStep,
+)
+
+from content_sync.pipelines.definitions.concourse.common.identifiers import (
+    OCW_STUDIO_WEBHOOK_RESOURCE_TYPE_IDENTIFIER,
+    OPEN_DISCUSSIONS_RESOURCE_IDENTIFIER,
+    SLACK_ALERT_RESOURCE_IDENTIFIER,
+)
+from content_sync.pipelines.definitions.concourse.common.image_resources import (
+    CURL_REGISTRY_IMAGE,
+)
+from main.utils import is_dev
+
+
+def add_error_handling(
+    step: Step, step_description: str, pipeline_name: str, instance_vars_query_str: str
+):
+    """
+    Add error handling steps to any Step-like object
+
+    Args:
+        step(Step): The Step-like object that uses StepModifierMixin to add the error handling steps to
+        step_description(str): A description of the step at which the failure occurred
+        instance_vars_query_str(str): A query string of the instance vars from the pipeline to build a URL with
+
+    Returns:
+        None
+    """
+    step_type = type(step)
+    if not issubclass(step_type, StepModifierMixin):
+        raise TypeError(
+            f"The step object of type {step_type} does not extend StepModifierMixin and therefore cannot have error handling"
+        )
+    concourse_base_url = settings.CONCOURSE_URL
+    concourse_team = settings.CONCOURSE_TEAM
+    concourse_path = (
+        f"/teams/{concourse_team}/pipelines/{pipeline_name}{instance_vars_query_str}"
+    )
+    concourse_url = urljoin(concourse_base_url, concourse_path)
+    on_failure_steps = [
+        OcwStudioWebhookStep(pipeline_name=pipeline_name, status="failed")
+    ]
+    on_error_steps = [
+        OcwStudioWebhookStep(pipeline_name=pipeline_name, status="errored")
+    ]
+    on_abort_steps = [
+        OcwStudioWebhookStep(pipeline_name=pipeline_name, status="errored")
+    ]
+    if not is_dev():
+        on_failure_steps.append(
+            SlackAlertStep(
+                alert_type="failed",
+                text=f"Failed - {step_description} : {concourse_url}",
+            )
+        )
+        on_error_steps.append(
+            SlackAlertStep(
+                alert_type="errored",
+                text=f"Concourse system error - {step_description} : {concourse_url}",
+            )
+        )
+        on_abort_steps.append(
+            SlackAlertStep(
+                alert_type="errored",
+                text=f"Concourse system error - {step_description} : {concourse_url}",
+            )
+        )
+    step.on_failure = TryStep(try_=DoStep(do=on_failure_steps))
+    step.on_error = TryStep(try_=DoStep(do=on_error_steps))
+    step.on_abort = TryStep(try_=DoStep(do=on_abort_steps))
```

Comment:
**Suggestion:** I think this should throw an error if `step` already has a failure/error/abort hook. (Or it could merge them, but I dunno if we need that now.)

Reply:
Yea, I think we could add `add_on_error_step` etc. functions later that would append a step to the `do` property, but the `add_error_handling` function was only meant to be called once so this makes sense.